Sources/OpenTelemetrySdk/Metrics/Stable/State/SynchronousMetricStorage.swift

```diff
@@ -76,7 +76,7 @@ public class SynchronousMetricStorage: SynchronousMetricStorageProtocol {
         
         var points = [PointData]()
         
-        aggregatorHandlesQueue.sync(flags: .barrier) {
+        aggregatorHandlesQueue.sync {
```

Comment:
I didn't do any load test here. Any concern here will require to convert to async but that will trigger more API changes